### DISCUSSION TITLE
RocketMQ4.9升级到rocketmq5.0后，线上业务怎么平滑升级啊，有方案吗？

### DISCUSSION THREAD

**Discussion by onejimmyboy** [apache/rocketmq#6325] [Q&A]
升级到rocketmq5.0后，之前4.9的应该还可以继续连接namesrv吗？还是需要连接proxy？

**Comment by zhouxinyu** [comment]
Proxy模块是可选的。不过使用5.0 gRPC协议SDK的话需要连接Proxy。

**Reply by onejimmyboy** [reply]
你指的是用这个clients连接的时候需要连接proxy？https://github.com/apache/rocketmq-clients

**Reply by drpmma** [reply]
是的，该repo的客户端基于gRPC协议实现，因此要链接proxy；同时proxy也分为local模式和cluster模式，可以认为local模式是原先broker namesrv架构的升级版。

**Comment by mxsm** [comment]
@onejimmyboy  升级到rocketmq5.0后如果你还是想用以前4.x的方式还是和之前的一样，但是如你想要用[RocketMQ Clients](https://github.com/apache/rocketmq-clients): gRPC/protobuf-based RocketMQ clients. 就需要部署proxy. RocketMQ Clients连接的是代理。
rocektmq里面的client继续连接namesrv和4.x一样。这里取决你的使用的客户端。官网的快速开始的例子就是使用的RocketMQ Clients + Proxy的模式。区别与4.x版本的例子

**Reply by onejimmyboy** [reply]
> @onejimmyboy 升级到rocketmq5.0后如果你还是想用以前4.x的方式还是和之前的一样，但是如你想要用[RocketMQ Clients](https://github.com/apache/rocketmq-clients): gRPC/protobuf-based RocketMQ clients. 就需要部署proxy. RocketMQ Clients连接的是代理。 rocektmq里面的client继续连接namesrv和4.x一样。这里取决你的使用的客户端。官网的快速开始的例子就是使用的RocketMQ Clients + Proxy的模式。区别与4.x版本的例子

哦哦明白了
